modules/oe_content_publication/oe_content_publication.post_update.php

```diff
@@ -34,39 +33,72 @@ function oe_content_publication_post_update_00002() {
 }
 
 /**
- * Create Contact reference field and update form display.
+ * Install modules and clear cached plugin definitions.
  */
 function oe_content_publication_post_update_00003() {
   $module_installer = \Drupal::service('module_installer');
   if (!\Drupal::service('module_handler')->moduleExists('oe_content_organisation_reference')) {
     $module_installer->install(['oe_content_organisation_reference']);
   }
-  $module_installer->install(['path', 'inline_entity_form']);
-
-  $storage = new FileStorage(drupal_get_path('module', 'oe_content_publication') . '/config/post_updates/00003_add_contact_reference');
+  $module_installer->install([
+    'composite_reference',
+    'entity_reference_revisions',
+    'field_group',
+    'inline_entity_form',
+    'link',
+    'path',
+    'oe_content',
+    'oe_content_departments_field',
+    'oe_content_documents_field',
+    'oe_content_entity_contact',
+    'oe_content_organisation_reference',
+    'oe_content_reference_code_field',
+    'oe_media_avportal',
+  ]);
 
   // Clear the cached plugin definitions of the field types.
   \Drupal::service('plugin.manager.field.field_type')->clearCachedDefinitions();
+}
+
+/**
+ * Create the new fields in the Publication content type.
+ */
+function oe_content_publication_post_update_00004(): void {
+  $storage = new FileStorage(drupal_get_path('module', 'oe_content_publication') . '/config/post_updates/00004_create_fields');
+  \Drupal::service('config.installer')->installOptionalConfig($storage);
+}
 
-  // Create the field storage for the reference field.
-  $field_storage_config = \Drupal::service('entity_type.manager')->getStorage('field_storage_config');
-  if (!$field_storage_config->load('node.oe_publication_contacts')) {
-    $reference_field = $storage->read('field.storage.node.oe_publication_contacts');
-    // We are creating the config which means that we are also shipping
-    // it in the config/install folder so we want to make sure it gets the hash
-    // so Drupal treats it as a shipped config. This means that it gets exposed
-    // to be translated via the locale system as well.
-    $reference_field['_core']['default_config_hash'] = Crypt::hashBase64(serialize($reference_field));
-    $field_storage_config->create($reference_field)->save();
+/**
+ * Update field settings.
+ */
+function oe_content_publication_post_update_00005(): void {
+  $new_field_labels = [
+    'node.oe_publication.oe_publication_type' => 'Resource type',
+    'node.oe_publication.oe_documents' => 'Files',
+    'node.oe_publication.title' => 'Title',
```

Comment:
Configuration `field.field.node.oe_publication.title` doesn't exist. Update is failed due to it.
Node title is in the `core.base_field_override.node.oe_publication.title`.

Reply:
This is fixed.